### PR TITLE
Stop asking for extensions if feature avaiable in core SPIR-V

### PR DESCRIPTION
**Pull request by antiagainst**
Migrating to unified grammar means we sometimes have two fields
for a certain feature: version and extensions. It means the feature
in question can be used either in SPIR-V of advanced-enough
versions or in any SPIR-V with with the specified extensions.

Validator now respects the above rules.

### CODE REVIEW

File: source/validate_instruction.cpp [hunk 1]
```diff
@@ -163,10 +163,15 @@ ExtensionSet RequiredExtensions(const ValidationState_t& state,
   if (state.grammar().lookupOperand(type, operand, &operand_desc) ==
       SPV_SUCCESS) {
     assert(operand_desc);
+    // If this operand is incorporated into core SPIR-V before or in the current
+    // target environment, we don't require extensions anymore.
+    if (static_cast<uint32_t>(state.context()->target_env) >=
```
Comment (dneto0) [inline source/validate_instruction.cpp]: Should use spvVersionForTargetEnv here.
Don't rely on enum orderings.

File: test/val/val_extensions_test.cpp [hunk 2]
```diff
@@ -228,4 +228,50 @@ TEST_P(ValidateAMDShaderBallotCapabilities, ExpectFailure) {
 INSTANTIATE_TEST_CASE_P(ExpectFailure, ValidateAMDShaderBallotCapabilities,
                         ValuesIn(AMDShaderBallotGroupInstructions()));
 
+struct ExtIntoCoreCase {
+  const char* cap;
```
Comment (dneto0) [inline test/val/val_extensions_test.cpp]: Should have some cases with Vulkan 1.0 and Vulkan 1.1.

File: source/validate_instruction.cpp (hunk 1 above)
Comment (antiagainst) [reply source/validate_instruction.cpp]: Very good catch! I've #1437 to solve it.

Comment (antiagainst) [reply source/validate_instruction.cpp]: Done

File: test/val/val_extensions_test.cpp (hunk 2 above)
Comment (antiagainst) [reply test/val/val_extensions_test.cpp]: Added tests for Vulkan target env.

**Comment by alegal-arm** [comment]
any ETA here?

**Comment by antiagainst** [comment]
@dneto0: I've addressed your comments. PTAL

**Comment by dneto0** [comment]
Rebased, squashed, and pushed into master as 43ca2112b8f

**Pull request closed without merging**